File: Back-End/src/main/java/iuh/fit/ecommerce/services/CartService.java
package iuh.fit.ecommerce.services;

import iuh.fit.ecommerce.dtos.request.cart.CartAddRequest;
import iuh.fit.ecommerce.dtos.request.cart.CartUpdateQuantityRequest;
import iuh.fit.ecommerce.dtos.response.cart.CartResponse;

public interface CartService {
    CartResponse getOrCreateCart();

    CartResponse addProduct(CartAddRequest request);

    CartResponse removeProduct(Long productVariantId);

    void clearCart(Long userId);

    CartResponse updateProductQuantity(CartUpdateQuantityRequest request);
}

File: Front-End/src/services/cart.service.ts
import axiosClient from "@/configurations/axios.config";
import type { CartAddRequest, CartResponse } from "@/types/cart.type";

export const cartService = {
  getCart: async () => {
    const response = await axiosClient.get<CartResponse>(`/carts`);
    return response.data;
  },
  addProductToCart: async (request: CartAddRequest) => {
    const response = await axiosClient.post<CartResponse>(`/carts/add`, request);
    return response.data;
  },

  removeProductFromCart: async (productVariantId: number) => {
    const response = await axiosClient.delete<CartResponse>(`/carts/remove/${productVariantId}`);
    return response.data;
  },

  updateCartItemQuantity: async (productVariantId: number, quantity: number) => {
    const response = await axiosClient.put<CartResponse>(`/carts/update-quantity`, {
      productVariantId,
      quantity
    });
    return response.data;
  },

  clearCart: async (userId: number) => {
    const response = await axiosClient.delete(`/carts/clear/${userId}`);
    return response.data;
  },
};
   